test/backend/server.test.tsx

```
@@ -0,0 +1,205 @@
+import assert from 'assert';
+import { createServer } from '../../src/backend/server';
+import * as dependencyToMock from '../../src/client/sdkGeneration';
+import { Plan, BuildStatus } from '../../src/model/Plan';
+import { Spec } from '../../src/model/Spec';
+
+/*
+ * Test services are registered and any hooks.
+ */
+describe('test server', () => {
+  let app;
+  beforeAll(async () => {
+    // TODO: Might need to change to beforeEach if data is stored and queried.
+    app = await createServer();
+  });
+
+  describe('test specification service', () => {
+    test('specification service registered', () => {
+      const s = app.service('specifications');
+      assert.ok(s, 'Registered the service');
+    });
+  });
+
+  describe('test plans service', () => {
+    let planData: Plan;
+    beforeEach(() => {
+      planData = {
+        specId: 1,
+        target: 'java is ew',
+        version: 'v1.0.0',
+        options: 'my options here',
+        buildStatus: BuildStatus.NOTRUN,
+      };
+    });
+
+    test('plans service registered', () => {
+      const s = app.service('plans');
+      assert.ok(s, 'Registered the service');
+    });
+
+    test('plan created', async () => {
+      const createdPlan = await app.service('plans').create(planData); // HTTP Create/
```

I'm pretty sure this should fail. specId 1 does not exist so it should throw an error.